backend/src/main/java/ch/puzzle/pcts/service/validation/TagValidationService.java

```diff
@@ -0,0 +1,17 @@
+package ch.puzzle.pcts.service.validation;
+
+import ch.puzzle.pcts.exception.PCTSException;
+import ch.puzzle.pcts.model.certificate.Tag;
+import ch.puzzle.pcts.model.error.ErrorKey;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Component;
+
+@Component
+public class TagValidationService {
+
+    public void validateName(Tag tag) {
+        if (tag.getName() == null) {
```

Comment:
this will result in a nullpointer and throws a 500 if tag is null in the first place. You should check whether tags is present or not first :D